packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -250,7 +282,28 @@ function WorklistDataGrid(props) {
   useEffect(() => {
     setDataGridColumns(displayedColumns.map((column) => initializeColumn(column)));
   // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [columns]);
+  }, [pinnedColumns, overflowColumns]);
+
+  // useEffect to calculate pinned column offsets
+  useEffect(() => {
+    if (pinnedColumns.length > 0) {
+      let cumulativeOffset = 0;
+      const offsetArray = [0];
+      const totalPinnedColumnsCount = hasSelectableRows ? pinnedColumns.length : pinnedColumns.length - 1;
```

Comment:
Why would the total pinned column count be the length of the pinned columns array minus 1?  Why do we need to subtract one?  Maybe a comment would help.

Reply:
This is because we are omitting the final pinned column from the offset arrays (since we only need to calculate the offset for the left edge of each column, and to calculate the edge of the last column, we only need the cumulative widths of all columns that came before it.).

Updated the variable name for clarity here: [b3b0076](https://github.com/cerner/terra-framework/pull/1710/commits/b3b00769ed5615dad6a7f14f771e655a44e05667)